npm install -D uses repository name instead of package name

Hi,
During the generation of a project, I am asked for both the name of the repo and the name of the npm package. For my current definitions, the repo is `peterolson/BigInteger.js` but the npm package is `big-integer` (not the same name), but the generator used the the repo name to download the module - and failed.

Note: I am using the latest generator available on npm (`0.14.10`)

Here are the logs:

````txt
D:\projects\typed\typed-big-integer>yo typings

     _-----_
    |       |    .--------------------------.
    |--(o)--|    |      Welcome to the      |
   `---------´   |    sensational typings   |
    ( _´U`_ )    |        generator!        |
    /___A___\    '--------------------------'
     |  ~  |
   __'.___.'__
 ´   `  |° ´ Y `

? What is the author/module of the source on github? peterolson/BigInteger.js
? Is the source installable through NPM? Yes
? Name of the package on NPM is... big-integer
? Is this module ambient? i.e. does it declare itself globally? No
? And your GitHub username is... demurgos
? Which license do you want to use? MIT
? Name to use on the license? demurgos
   create package.json
   create .vscode\settings.json
   create test\tsconfig.json
   create source-test\README.md
   create source-test\tsconfig.json
   create index.d.ts
   create tsconfig.json
   create tslint.json
   create .editorconfig
   create .travis.yml
   create .gitignore
   create .npmignore
   create typings.json
   create README.md
   create test\test.ts
   create LICENSE
Installing BigInteger.js...
npm ERR! Windows_NT 10.0.10240
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-D" "--save-exact" "BigInteger.js"
npm ERR! node v5.1.0
npm ERR! npm  v3.3.12
npm ERR! code E404

npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/BigInteger.js
npm ERR! 404
npm ERR! 404 'BigInteger.js' is not in the npm registry.
npm ERR! 404 Your package name is not valid, because
npm ERR! 404
npm ERR! 404  1. name can no longer contain capital letters
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\projects\typed\typed-big-integer\npm-debug.log
Downloading peterolson/BigInteger.js...
Initialized empty Git repository in D:/projects/typed/typed-big-integer/.git/
Cloning into 'source'...
remote: Counting objects: 872, done.
rRemote: Total 872 (delta 0), reused 0 (delta 0), pack-reused 872
Receiving objects: 100% (872/872), 433.34 KiB | 346.00 KiB/s, done.
Resolving deltas:   0% (0/538)
Resolving deltas: 100% (538/538), done.
Checking connectivity... done.
warning: LF will be replaced by CRLF in .gitmodules.
The file will have its original line endings in your working directory.

I am done! Now it is your turn!

If DefinitelyTyped has a definition for the source,
 you can run tsd install <source> to download the file
 so you can easily access those definitions.
You don't need to save it though.

Run npm install to install dependencies needed. It will take about a minute.
In the meantime, you can open your editor and start writing.

When installation completes, run npm run watch.
It will automatically build and test the typings for you as you make changes.
````

--- a/generators/current/index.js
+++ b/generators/current/index.js
@@ -212,8 +212,8 @@ module.exports = yeoman.Base.extend({
   install: {
     npmInstallSource() {
       if (this.npmName) {
-        this.log(`Installing ${chalk.green(this.sourcePackageName)}...`);
-        this.spawnCommandSync('npm', ['install', '-D', '--save-exact', this.sourcePackageName]);
+        this.log(`Installing ${chalk.green(this.npmName)}...`);
+        this.spawnCommandSync('npm', ['install', '-D', '--save-exact', this.npmName]);
       }
     },
     submodule() {

--- a/package.json
+++ b/package.json
@@ -22,8 +22,6 @@
   "dependencies": {
     "chalk": "^1.0.0",
     "change-case": "^2.3.1",
-    "github": "^0.2.4",
-    "nodegit": "^0.12.1",
     "yeoman-generator": "^0.22.5",
     "yosay": "^1.1.0"
   },

